adapters/algorix/algorix.go

```diff
@@ -0,0 +1,171 @@
+package algorix
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/prebid/prebid-server/errortypes"
+	"github.com/prebid/prebid-server/macros"
+	"net/http"
+	"text/template"
+
+	"github.com/mxmCherry/openrtb/v15/openrtb2"
+	"github.com/prebid/prebid-server/adapters"
+	"github.com/prebid/prebid-server/config"
+	"github.com/prebid/prebid-server/openrtb_ext"
+)
+
+type Adapter struct {
+	EndpointTemplate template.Template
+}
+
+// Builder builds a new instance of the Foo adapter for the given bidder with the given config.
+func Builder(bidderName openrtb_ext.BidderName, config config.Adapter) (adapters.Bidder, error) {
+	endpoint, err := template.New("endpointTemplate").Parse(config.Endpoint)
+	if err != nil {
+		return nil, fmt.Errorf("unable to parse endpoint url template: %v", err)
+	}
+	bidder := &Adapter{
+		EndpointTemplate: *endpoint,
+	}
+	return bidder, nil
+}
+
+// MakeRequests Make Requests
+func (adapter *Adapter) MakeRequests(request *openrtb2.BidRequest, requestInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+	var adapterRequests []*adapters.RequestData
+	var errs []error
+
+	adapterRequest, err := adapter.makeRequest(request)
+	if err == nil {
+		adapterRequests = append(adapterRequests, adapterRequest)
+	} else {
+		errs = append(errs, err)
+	}
+	return adapterRequests, errs
+}
+
+func (adapter *Adapter) makeRequest(request *openrtb2.BidRequest) (*adapters.RequestData, error) {
+	if len(request.Imp) == 0 {
+		return nil, &errortypes.BadInput{Message: "No impression in the request"}
+	}
+
+	algorixExt := parseAlgoriXExt(request)
+
+	if algorixExt == nil {
+		return nil, &errortypes.BadInput{Message: "Invalid ExtImpAlgoriX value"}
+	}
+
+	endPoint, err := adapter.getEndPoint(algorixExt)
+	if err != nil {
+		return nil, err
+	}
+
+	preProcess(request)
+	reqBody, err := json.Marshal(request)
+	if err != nil {
+		return nil, err
+	}
+
+	headers := http.Header{}
+	headers.Add("Content-Type", "application/json;charset=utf-8")
+	headers.Add("Accept", "application/json")
+	headers.Add("x-openrtb-version", "2.5")
+
+	return &adapters.RequestData{
+		Method:  "POST",
+		Uri:     endPoint,
+		Body:    reqBody,
+		Headers: headers,
+	}, nil
+}
+
+//parseAlgoriXExt parse AlgoriX Ext
+func parseAlgoriXExt(request *openrtb2.BidRequest) *openrtb_ext.ExtImpAlgorix {
+	var extImpAlgoriX openrtb_ext.ExtImpAlgorix
+	for _, imp := range request.Imp {
+		var extBidder adapters.ExtImpBidder
+		err := json.Unmarshal(imp.Ext, &extBidder)
+		if err != nil {
+			continue
+		}
+		err = json.Unmarshal(extBidder.Bidder, &extImpAlgoriX)
+		if err != nil || len(extImpAlgoriX.Sid) == 0 || len(extImpAlgoriX.Token) == 0 {
```

Comment:
File `static/bidder-params/algorix.json` defines a `"minLength": 1` for both parameters which makes prebid server core to discard `Imp[i].Ext` entries whenever these two fields are empty. Therefore, we can discard these two checks:
```
91         err = json.Unmarshal(extBidder.Bidder, &extImpAlgoriX)
92 -       if err != nil || len(extImpAlgoriX.Sid) == 0 || len(extImpAlgoriX.Token) == 0 {
   +       if err != nil {
93             continue
94         }
adapters/algorix/algorix.go
```